lab-justin/server.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const http = require('http');
+const port = process.env.PORT || 3000;
+const parseUrl = require('url').parse;
+const parseQuery = require('querystring').parse;
+const bodyParser = require('./lib/bodyParser');
+const cowsay = require('cowsay');
+
+const server = http.createServer(function(req, res){
+  req.url = parseUrl(req.url);
+  // console.log('req.url', req.url);
+  req.url.query = parseQuery(req.url.query);
+  // console.log('req.url.query', req.url.query);
+  console.log('req.method', req.method);
+
+  if (req.url.pathname === '/'){
+    res.writeHead(200, {'Content-Type': 'text/plain'});
+    res.write('API Endpoints: \n /api/cowsay');
+    return res.end();
+  }
+
+  if(req.method === 'GET' && req.url.pathname ==='/api/cowsay'){
+    bodyParser(req).then(function(){
+      res.writeHead(200, {'Content-Type': 'text/plain'});
+      // res.write('OK');
+      res.write(cowsay.say({
+        text: 'mooo~',
+        e:'o*',
+        T:'U'
+      }));
+      return res.end();
+    })
+    .catch(function(err){
```

Comment:
As is, this only runs when there is an error, but the assignment says that it should also run if no query string is provided specifying the text.